presto-main/src/main/java/io/prestosql/sql/planner/NullityEvaluator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner;
+
+import io.prestosql.sql.tree.ComparisonExpression;
+import io.prestosql.sql.tree.DefaultExpressionTraversalVisitor;
+import io.prestosql.sql.tree.Expression;
+import io.prestosql.sql.tree.Literal;
+import io.prestosql.sql.tree.LogicalBinaryExpression;
+import io.prestosql.sql.tree.Node;
+import io.prestosql.sql.tree.NullLiteral;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Determines whether a given Expression can return null value when evaluated
+ */
+public final class NullityEvaluator
+{
+    private NullityEvaluator() {}
+
+    public static boolean canEvaluateToNull(Expression expression)
+    {
+        requireNonNull(expression, "expression is null");
+        Boolean re = new Visitor().process(expression, null);
+        return re == null || re;
+    }
+
+    private static class Visitor
+            extends DefaultExpressionTraversalVisitor<Boolean, Void>
```

Comment:
This is just a initial version and its almost equivalent to `node instanceof Literal && !(node instanceofNull)` because only for a literal (except `NullLiteral` we can assure that it can't evaluate to null) , the next implementation is to analyze based on the function and the filter nodes through which symbol is passed i.e if a symbol columnA passes though a `FilterNode` having its filter criteria as `columnA > 10` after the filter operation is performed we can assure that `columnA` won't have any null values (assuming there isn't any Join operations). This is just a concrete idea need to work on the actual implementation. So instead of using 
 `node instanceof Literal && !(node instanceofNull)`  we tried going for an evaluator like a `DeterminismEvaluator`